src/sparsezoo/objects/model_folder_objects.py

```diff
@@ -0,0 +1,442 @@
+    :param path: path of the SampleOriginals
+    :param url: url of the SampleOriginals
+    """
+
+    def __init__(
+        self,
+        files: List[File],
+        name: str,
+        path: Optional[str] = None,
+        url: Optional[str] = None,
+    ):
+        super().__init__(files=files, name=name, path=path, url=url)
+
+    def validate(self, integration: Optional[str] = None) -> bool:
+        """
+        Validate whether all the sample originals files adhere to
+        the convention imposed by the integration.
+
+        :param integration: name of the specific integration
+            (e.g. transformers, YOLOv5 etc.)
+        :return: boolean flag; True if files are valid and no errors arise
+        """
+        # TODO: Assuming for now that all files are loadable
+        for file in self.files:
+            file.validate()
+
+        return True
+
+    def __iter__(self):
+        for file in self.files:
+            yield file
+
+
+class NumpyDirectory(Directory):
+    """
+    Object that represents a directory with unedited data
+    that can be used as inputs to a training pipeline.
+
+    :param files: list of files contained within the NumpyDirectory
+    :param name: name of the NumpyDirectory
+    :param path: path of the NumpyDirectory
+    :param url: url of the NumpyDirectory
+    """
+
+    def __init__(
+        self,
+        files: List[File],
+        name: str,
+        path: Optional[str] = None,
+        url: Optional[str] = None,
+    ):
+        super().__init__(files=files, name=name, path=path, url=url)
+
+    def validate(
+        self,
+        model: Optional[onnx.ModelProto] = None,
+        integration: Optional[str] = None,
+        strict_mode: bool = True,
+    ) -> bool:
+        """
+        Validate whether all the files in numpy directory
+        adhere to the convention imposed by the integration.
+
+        :param model: ONNX model
+        :param integration: integration: name of the specific integration
+            (e.g. transformers, YOLOv5 etc.)
+        :param strict_mode:
+            - if strict_mode: validation will raise ValueError on error
+            - if not strict_mode: validation will raise warning on error
+        :return: boolean flag; True if files are valid and no errors arise
+        """
+        for file in self.files:
+            file._validate_numpy(strict_mode=strict_mode)
+
+        if model:
+            self._validate_model(model)
+
+        return True
+
+    def _validate_model(self, model: Optional[onnx.ModelProto]):
+        input_names = [inp.name for inp in model.graph.input]
+        for file in self.files:
+            numpy_list = load_numpy_list(file.path)
+            for numpy_dict in numpy_list:
+                if input_names != list(numpy_dict.keys()):
+                    raise ValueError(
+                        "The keys in npz dictionary do not match "
+                        "the input names of the provided model!"
```

Comment:
nit: remove `!`, include names of found keys and expected keys with f-string

Reply:
Adding it to the new PR in the other branch.